cpp/include/cudf/fixed_point/floating_conversion.hpp

```diff
@@ -609,6 +640,641 @@ CUDF_HOST_DEVICE inline constexpr T divide_power10(T value, int exp10)
+                        int exp2,
+                        int exp10)
+{
+  // This is similar to shift_to_decimal_posexp(), except exp10 < 0 & exp2 < 0
+  // See comments in that function for details.
+  // Instead here we need to multiply by 10s and shift right by 2s
+
+  // Convert to using positive values so we don't have keep negating each time we multiply
+  int exp2_mag  = -exp2;
+  int exp10_mag = -exp10;
+
+  // ShiftingRep: uint64 for float's, __uint128_t for double's
+  using Constants   = shifting_constants<FloatingType>;
+  using ShiftingRep = typename Constants::ShiftingRep;
+  ShiftingRep shifting_rep;
+
+  // For performing final 10s-shift
+  auto final_shifts_low10s = [&]() {
+    // Last 10s-shift: multiply all remaining decimal places, shift all remaining bits, then bail
+    // The multiplier is less than the max-shift, and thus fits within 64 / 32 bits
+    if constexpr (Constants::is_double) {
+      shifting_rep = multiply_power10_64bit(shifting_rep, exp10_mag);
+    } else {
+      shifting_rep = multiply_power10_32bit(shifting_rep, exp10_mag);
+    }
+
+    // Final bit shift: Shift may be large, guard against UB
+    return guarded_right_shift(shifting_rep, exp2_mag);
+  };
+
+  // If our total decimal shift is less than the max, we don't need to iterate
+  if (exp10_mag <= Constants::max_digits_shift) {
+    shifting_rep = base2_value;
+    return final_shifts_low10s();
+  }
+
+  // We want to start by lining up our bits to num_rep_bits, but since we'll be bit-shifting
+  // down, we need even more low bits as a buffer (see comments on these constants)
+  auto const lineup_shift       = Constants::num_rep_bits - count_significant_bits(base2_value);
+  auto const num_init_bit_shift = lineup_shift + Constants::num_2s_shift_buffer_bits;
+
+  // Constants::num_2s_shift_buffer_bits; Note: This shift is safe to do in the smaller IntegerRep
+  // as it is up to bit 61 / 31
+  shifting_rep = base2_value << num_init_bit_shift;
+  exp2_mag += num_init_bit_shift;
+
+  // Iterate, multiplying by 10s and shifting down by 2s until we're almost done
+  do {
+    // More decimal places to shift than we have room: Multiply the max number of 10s
+    shifting_rep *= Constants::max_digits_shift_pow;
+    exp10_mag -= Constants::max_digits_shift;
+
+    // If our remaining bit shift is less than the max, we're finished iterating
+    if (exp2_mag <= Constants::max_bits_shift) {
+      // Last bit-shift: Shift all remaining bits, apply the remaining scale, then bail
+      shifting_rep >>= exp2_mag;
+
+      // We need to convert to the output rep for the final scale-factor multiply, because if (e.g.)
+      // float -> dec128 and some large exp10_mag, it might overflow the 64bit shifting rep.
+      // It's not needed for exp10 > 0 because we're dividing by 10s there instead of multiplying.
+      using UnsignedRep = cuda::std::make_unsigned_t<Rep>;
+      // NOTE: This can overflow! (Both multiply and cast)
```

Comment:
It is a problem, it's a TODO for a potential future PR, I'm just trying to call out everywhere it can happen.  This is a problem of the current conversion anyway so this isn't a new problem only introduced here.